db/migrate/20150921090740_add_index_to_domain_name.rb

```diff
@@ -0,0 +1,16 @@
+class AddIndexToDomainName < ActiveRecord::Migration
+  def up
+    case ActiveRecord::Base.connection.instance_values["config"][:adapter]
+    when "postgresql"
+      execute 'CREATE UNIQUE INDEX domain_name_index ON domains (lower(name))'
+    when "sqlite3"
+      execute 'CREATE UNIQUE INDEX domain_name_index ON domains (name collate nocase)'
+    else
+      add_index :domains, :name, :name => 'domain_name_index', :unique => true
```

Comment:
If this relies on MySQL unique indexes being case sensitive by default, please leave a comment